### PR TITLE
improve: LDP-2009: Make checking canonical URLs more flexible

### PR DESCRIPTION
**Pull request by fago**

### CODE REVIEW

File: tests/helpers/test-methods.js [hunk 1]
```diff
@@ -37,37 +37,42 @@ module.exports = {
...
       }
     }
-    expect(canonicalExists, 'Expect canonical url to exist').toBeTruthy();
+    expect(canonicalExists, 'Expect canonical URL to exist').toBeTruthy();
   },
-  checkMetaUrlContentByProperty: async ([response, key, value, frontendURL]) => {
+  /**
+   * @deprecated
+   */
+  checkHomepageCanonicalUrls: async ([response, frontendURL]) => {
```
Comment (fago) [inline tests/helpers/test-methods.js:61]: diff is weird. the deprecated method is not changed

File: tests/helpers/test-methods.js [hunk 2]
```diff
@@ -37,37 +37,42 @@ module.exports = {
...
   },
-  checkHomepageCanonicalUrls: async ([response, frontendURL]) => {
+  /**
+   * Checks the canonical URL within the given CE-API response.
+   *
+   * @param {Response} response The given CE-API response
+   * @param canonicalURL The expected URL
+   * @returns {Promise<void>}
+   */
+  expectCanonicalUrl: async (response, canonicalURL) => {
```
Comment (petarbasic) [inline tests/helpers/test-methods.js:47]: I had issues in the past when having the function arguments passed like this. Can't remember why, but it only worked if it was passed as an array. See other funcs with multiple arguments. 

Maybe this works now, has it been tested like this?

Comment (fago) [reply tests/helpers/test-methods.js:47]: yes, I was wondering about that weird array syntax. Yes, I tested it, works fine, so let's better start with regular params?

**Review by petarbasic — APPROVED** [review]

File: tests/helpers/test-methods.js (hunk 2 above)
Comment (fago) [reply tests/helpers/test-methods.js:47]: @petarbasic yes, I tested

**Pull request merged**